### PR TITLE
black formatting

### PR DESCRIPTION
**Pull request by max-sixty**
I found this v good in other small projects, but obv your call @shoyer

I also dropped Py2 & Py3.4 testing and added 3.7

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,4 +30,7 @@ install:
   - python setup.py install
 
 script:
-  - pytest
+  - if [[ "$TASK" == "lint" ]]; then
+      pip install black && black .
```
Comment (shoyer) [inline .travis.yml]: There's a syntax here in your bash script. I think you may need to add semicolins at the end of each line?

File: numbagg/moving.py [hunk 2]
```diff
@@ -6,48 +6,47 @@
...
             if is_observation:
                 old_wt *= old_wt_factor
 
                 # avoid numerical errors on constant series
                 if weighted_avg != cur:
-                    weighted_avg = ((old_wt * weighted_avg) +
-                                    (new_wt * cur)) / (old_wt + new_wt)
+                    weighted_avg = ((old_wt * weighted_avg) + (new_wt * cur)) / (
+                        old_wt + new_wt
+                    )
```
Comment (shoyer) [inline numbagg/moving.py]: this is the only line that looks actively bad to me with this new formatting

File: numbagg/transform.py [hunk 3]
```diff
@@ -45,15 +45,14 @@ def _apply_ast_rewrite(func, node_transformer):
...
     exec(source, _func_globals(func), scope)
     try:
         return scope[_TRANFORMED_FUNC_NAME]
     except KeyError:
-        raise TypeError('failed to rewrite function definition:\n%s'
-                        % orig_source)
+        raise TypeError("failed to rewrite function definition:\n%s" % orig_source)
 
 
 def _ast_arg(name):
```
Comment (shoyer) [inline numbagg/transform.py]: we can clean up this function if we don't care about Python 2 support (and we definitely do not care about Python 2 anymore!)

**Comment by max-sixty** [comment]
Green!

**Comment by shoyer** [comment]
OK, if one case of bad formatting is the price of not needing to think about it again... we can give this a try!

File: numbagg/moving.py (hunk 2 above)
Comment (max-sixty) [reply numbagg/moving.py]: I agree

**Pull request merged**